platform-hub-web/src/app/shared/auth/auth.service.js

```diff
@@ -99,16 +90,7 @@ export const authService = function ($window, $cookies, $q, $filter, $base64, $s
     }
   }
 
-  function windowPopupOptions() {
-    return {
-      width: popupHeight,
-      height: popupWidth,
-      top: $window.screenY + (($window.outerHeight - popupHeight) / 2.5),
-      left: $window.screenX + (($window.outerWidth - popupWidth) / 2)
-    };
-  }
-
   function broadcastAuthData() {
-    $rootScope.$broadcast(authDataEvent, getPayload());
+    $rootScope.$broadcast(events.auth.updated, getPayload());
```

Comment:
Agreed it's not good practice _in general_ – when overused and when being too liberal about what data updates get shared using such global events.

In this particular case, the two events I've defined are _intended_ to share the two global objects across the app (auth status and "me" profile data) so that any component that relies on them can react accordingly. I'd certainly question it's use beyond these two cases (since any further data updates should most likely be isolated to a particular section / component, and maybe it's children). At that point I'd start using something like [ng-redux](https://github.com/angular-redux/ng-redux).

In terms of these data objects being global, if any component/service wants to derive data or augment them then they should clone them first (this is a general rule for any part of the app really, to keep things a bit more functional).

How would refactoring it into a model help in this case? (Note that I need to be able to listen out for changes from various different places in the app, and would rather not use `$scope.$watch(() => { monitor_service_method_for_changes }, () => { … })`).